pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/PartitionedConsumerImpl.java

```diff
@@ -398,6 +398,14 @@ public boolean isBatchingAckTrackerEmpty() {
         return consumers;
     }
 
+    public int getAvailablePermits() {
```

Comment:
The interface for these 2 methods could also be added to `ConsumerBase` abstract class, to avoid need to downcast and checking the type of the consumer implementation